File: string/2000_reverse_prefix_of_word/2000_reverse_prefix_of_word.go
package main

func reversePrefix(word string, ch byte) string {
	index := -1
	for i := 0; i < len(word); i++ {
		if word[i] == ch {
			index = i
			break
		}
	}
	if index == -1 {
		return word
	}
	res := make([]byte, 0, len(word))
	for i := index; i >= 0; i-- {
		res = append(res, word[i])
	}
	for i := index + 1; i < len(word); i++ {
		res = append(res, word[i])
	}
	return string(res)
}

File: string/2000_reverse_prefix_of_word/2000_reverse_prefix_of_word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        result = ""
        index = len(word)
        for idx, c in enumerate(word):
            if c == ch:
                index = idx
                break
        if index == len(word):
            return word
        
        for i in range(index, -1, -1):
            result += word[i]
        return result + word[index+1:]